### PR TITLE
upd(comprehensive-demo-react to comprehensive-demo-react16):add(comprehensive-demo-react18 sample):e2e(comprehensive-demo-react16 &&  comprehensive-demo-react18 samples)

### PR DESCRIPTION
**Pull request by PeaceDate**
comprehensive-demo-react16
- [x] Update sample name to comprehensive-demo-react16
- [x] Add test for sample
Screenshot 2022-12-28 at 12 00 22

comprehensive-demo-react18
- [x]  Create new sample with react 18 version from exiting comprehensive demo sample
- [x]  Add tests for this sample
Screenshot 2022-12-28 at 11 59 59

### CODE REVIEW

**Comment by ScriptedAlchemy** [comment]
neeeds rebase

**Comment by ScriptedAlchemy** [comment]
needs rebase

File: comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+            selector: baseSelectors.linkTag,
+            index: Constants.hrefs.comprehensiveDemoRoutingDemoLink.index,
+            attr: Constants.commonText.attr,
+            prop: Constants.commonText.href,
+            value: Constants.hrefs.comprehensiveDemoRoutingDemoLink.link
+        })
+    })
+
+    it('Check Dialog elements', () => {
+        basePage.openLocalhost(3001, '#/dialog')
```
Comment (edchai) [inline comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts]: link to constant

File: comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            text: Constants.elementsText.comprehensiveDemoDialogButtonTextApp2
+        })
+        basePage.clickElementWithText({
+            selector: baseSelectors.button,
+            text: Constants.elementsText.comprehensiveDemoDialogButtonTextApp2
+        })
+    })
+
+    it('Check Svelte Page elements', () => {
+        basePage.openLocalhost(3001, '#/svelte')
```
Comment (edchai) [inline comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts]: same

File: comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+            text: Constants.commonPhrases.standartText
+        })
+        basePage.checkElementWithTextPresence({
+            selector: baseSelectors.h1,
+            text: Constants.elementsText.comprehensiveDemoSvelteUpdatedText
+        })
+    })
+
+    it('Check Routing elements', () => {
+        basePage.openLocalhost(3001, '#/routing/foo')
```
Comment (edchai) [inline comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts]: same

File: comprehensive-demo/e2e/checkComrehensiveDemoApp2.cy.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import { dialogs } from './../../cypress/common/selectors';
+import { BaseMethods } from "../../cypress/common/base";
+import { block, baseSelectors } from "../../cypress/common/selectors";
+import { Constants } from "../../cypress/fixtures/constants";
+
+const basePage: BaseMethods = new BaseMethods()
+
+describe('Check Comprehensive Demo App2', () => {
```
Comment (edchai) [inline comprehensive-demo/e2e/checkComrehensiveDemoApp2.cy.ts]: add some more description. There are a lot of checks for such vague description

File: comprehensive-demo/e2e/checkComrehensiveDemoApp3.cy.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import { BaseMethods } from "../../cypress/common/base";
+
+const basePage: BaseMethods = new BaseMethods()
+
+describe('Check Comprehensive Demo App3', () => {
+    beforeEach(() => {
+        basePage.openLocalhost(3003)
+    })
+
+    it('Check App build and running', () => {
```
Comment (edchai) [inline comprehensive-demo/e2e/checkComrehensiveDemoApp3.cy.ts]: same

File: cypress/common/base.ts [hunk 6]
```diff
@@ -568,5 +573,44 @@ export class BaseMethods {
         expect(text).to.be.eq(value)
 
     }
+
+    public checkBrowserAlertByText({
```
Comment (edchai) [inline cypress/common/base.ts:624]: this method should be already added to base. Please check for duplicate

File: cypress/common/selectors.ts [hunk 7]
```diff
@@ -100,14 +103,25 @@ export const buttons = {
...
 
 export const alertMessages = {
-    angularAlertMessage: '.alert-danger'
+    angularAlertMessage: '.alert-danger',
+    alertBox: 'alert-box',
+    alert: '.alert'
 }
 
 export const block = {
+    comprehensiveDemoBlockFirstApp3: '.makeStyles-root-1',
```
Comment (edchai) [inline cypress/common/selectors.ts]: you can merge comprehensiveDemoBlock selectors to object

File: cypress/fixtures/constants.ts [hunk 8]
```diff
@@ -70,6 +70,59 @@ export class Constants {
         diferentReactVersionsApp1ReactBlockSubheader: 'And these are children passed into it from the legacy app',
         diferentReactVersionsParagraph: 'More react components from App2 using non-legacy React to render',
         diferentReactVersionsButtonName: 'App 2 Button',
+        comprehensiveDemoButtonTextApp5: 'bar',
```
Comment (edchai) [inline cypress/fixtures/constants.ts]: please recheck all added constants and check what you can merge to objects

File: cypress/fixtures/constants.ts [hunk 9]
```diff
@@ -70,6 +70,59 @@ export class Constants {
...
+        comprehensiveDemoAlertTextApp5: 'Hello',
+        comprehensiveDemoAlertMessage: 'You have pressed a button.',
+        comprehensiveDemoHeaderTextApp4: 'Hello From Svelte world!',
+        comprehensiveDemoHeaderTextApp3: 'Styled Components App',
+        comprehensiveDemoHeaderButtonTextApp3: '💅 Test Button',
```
Comment (edchai) [inline cypress/fixtures/constants.ts]: constant with such text should be already added

**Review by edchai — CHANGES_REQUESTED** [review]

File: comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts (hunk 1 above)
Comment (PeaceDate) [reply comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts]: Fixed

File: comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts (hunk 2 above)
Comment (PeaceDate) [reply comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts]: Fixed

File: comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts (hunk 3 above)
Comment (PeaceDate) [reply comprehensive-demo/e2e/checkComrehensiveDemoApp1.cy.ts]: Fixed

File: comprehensive-demo/e2e/checkComrehensiveDemoApp2.cy.ts (hunk 4 above)
Comment (PeaceDate) [reply comprehensive-demo/e2e/checkComrehensiveDemoApp2.cy.ts]: Fixed

File: comprehensive-demo/e2e/checkComrehensiveDemoApp3.cy.ts (hunk 5 above)
Comment (PeaceDate) [reply comprehensive-demo/e2e/checkComrehensiveDemoApp3.cy.ts]: Fixed

File: cypress/common/base.ts (hunk 6 above)
Comment (PeaceDate) [reply cypress/common/base.ts:624]: Fixed

File: cypress/common/selectors.ts (hunk 7 above)
Comment (PeaceDate) [reply cypress/common/selectors.ts]: Fixed

File: cypress/fixtures/constants.ts (hunk 8 above)
Comment (PeaceDate) [reply cypress/fixtures/constants.ts]: Fixed

File: cypress/fixtures/constants.ts (hunk 9 above)
Comment (PeaceDate) [reply cypress/fixtures/constants.ts]: Fixed

File: cypress/common/base.ts (hunk 6 above)
Comment (edchai) [reply cypress/common/base.ts:624]: You deleted method with increased wait for alert. Please increase to 3000 in which you left

File: cypress/common/base.ts [hunk 10]
```diff
@@ -568,5 +549,47 @@ export class BaseMethods {
...
+                }
+                resolve();
+            });
+            setTimeout(() => {
+                reject(new Error('window.confirm wasn\'t called within 1s'));
+            }, 1000);
+        }), { log: false });
+    }
+
+    public checkUrl(path: string): void {
```
Comment (edchai) [inline cypress/common/base.ts]: we already have checkUrlText

File: cypress/common/base.ts [hunk 11]
```diff
@@ -568,5 +549,47 @@ export class BaseMethods {
...
+        cy.url().should('contain', path)
+    }
+
+    public returnBack(): void {
```
Comment (edchai) [inline cypress/common/base.ts]: already have goBack

**Review by edchai — CHANGES_REQUESTED** [review]

File: cypress/common/base.ts (hunk 10 above)
Comment (PeaceDate) [reply cypress/common/base.ts]: Fixed

File: cypress/common/base.ts (hunk 11 above)
Comment (PeaceDate) [reply cypress/common/base.ts]: Fixed

**Review by edchai — APPROVED** [review]

**Pull request merged**